src/hotspot/os/posix/os_posix.cpp

```diff
@@ -1367,6 +1281,48 @@ void os::Posix::to_RTC_abstime(timespec* abstime, int64_t millis) {
              true  /* use real-time clock */);
 }
 
+// Common (partly) shared time functions
+
+jlong os::javaTimeMillis() {
+  struct timespec ts;
+  int status = clock_gettime(CLOCK_REALTIME, &ts);
+  assert(status == 0, "clock_gettime error: %s", os::strerror(errno));
+  return jlong(ts.tv_sec) * MILLIUNITS +
+    jlong(ts.tv_nsec) / NANOUNITS_PER_MILLIUNIT;
+}
```

Comment:
David,
Great to see simplifications like this happening in OpenJdk code.
- Posix letting you delete platform specific code.
- Always nice to see 'if' branching eliminated to simplify test coverage and runtime.